src/hookproxy/proxydll.cpp

```diff
@@ -0,0 +1,251 @@
+}
+__declspec(naked) void FakeAVIFileInit()
+{
+    _asm { jmp[OriginalAVIFileInit] }
+}
+__declspec(naked) void FakeAVIFileOpenA()
+{
+    _asm { jmp[OriginalAVIFileOpenA] }
+}
+__declspec(naked) void FakeAVIFileRelease()
+{
+    _asm { jmp[OriginalAVIFileRelease] }
+}
+__declspec(naked) void FakeAVIFileExit()
+{
+    _asm { jmp[OriginalAVIFileExit] }
+}
+__declspec(naked) void FakeAVIFileCreateStreamA()
+{
+    _asm { jmp[OriginalAVIFileCreateStreamA] }
+}
+__declspec(naked) void FakeAVIStreamSetFormat()
+{
+    _asm { jmp[OriginalAVIStreamSetFormat] }
+}
+__declspec(naked) void FakeAVIStreamRelease()
+{
+    _asm { jmp[OriginalAVIStreamRelease] }
+}
+
+/**
+ * Get the path for the exe we were loaded by.
+ */
+wchar_t *Get_Path()
+{
+    wchar_t buffer[MAX_PATH];
+    wchar_t *path = nullptr;
+    wchar_t *full_path = nullptr;
+
+    while (true) {
+        DWORD size;
+
+        size = GetModuleFileNameW(nullptr, buffer, sizeof(buffer) / sizeof(buffer[0]));
+
+        if (size == 0) {
+            break;
+        } else if (size == (DWORD)(sizeof(buffer) / sizeof(buffer[0]))) {
+            DWORD size_ = size;
+            do {
+                wchar_t *path_ = (wchar_t *)std::realloc(path, sizeof(wchar_t) * size_ * 2);
+
+                if (path_ == nullptr) {
+                    break;
+                }
+
+                size_ *= 2;
+                path = path_;
+                size = GetModuleFileNameW(nullptr, path, size_);
+            } while (size == size_);
+
+            if (size == size_) {
+                break;
+            }
+        } else {
+            path = buffer;
+        }
+
+        full_path = _wfullpath(nullptr, path, MAX_PATH);
+
+        break;
+    }
+
+    if (path != buffer) {
+        free(path);
+    }
+
+    return full_path;
+}
+
+/**
+ * Get the hash for the exe we were loaded by.
+ */
+const char* Get_Hash()
+{
+    static char hash[41];
+    std::memset(hash, 0, sizeof(hash));
+
+    SHAEngine sha;
+    char buffer[1024];
+    wchar_t *path = Get_Path();
+
+    if (path != nullptr) {
+        std::ifstream in(path, std::ios_base::in | std::ios_base::binary);
+        free(path);
+
+        // Get the hash of the binary.
+        do {
+            in.read(buffer, sizeof(buffer));
+            sha.Hash(buffer, int(in.gcount()));
+        } while (in.gcount() == sizeof(buffer));
+
+        sha.Print_Result(hash);
+    }
+
+    return hash;
+}
+
+/**
+ * Get the payload dll name we expect for the binary based on its hash.
+ */
+const char* Get_Payload_Name()
+{
+    //Exe hashes we support and the payload dll name we expect for them.
+    static const struct ExePayloads
+    {
+        const char *exe_hash;
+        const char *payload_name;
+    } Payloads[] = {
+        { "5b805b4ecba79581fe8ec454b5acf7b7e13d4dd5", "thyme.dll" },
+        { nullptr, nullptr },
+    };
+
+    const char *target_hash = Get_Hash();
+
+    for (unsigned i = 0; Payloads[i].exe_hash != nullptr; ++i) {
```

Comment:
Loop can just condition with `i < ARRAY_SIZE(Payloads)`. There is no need for a null terminated array here.